### ISSUE TITLE
No cursor themes in Xfce

### ISSUE THREAD

**Issue by ambrop72** [NixOS/nixpkgs#21442]
## Issue description

In the Xfce "Mouse and Touchpad" settings, in the Theme tab, no cursor themes are shown other than Default. The cursor size option has no effect. Even after installing `pkgs.vanilla-dmz` (a set of cursor themes) via `environment.systemPackages`, no cursor themes are available.

## Technical details

* System: NixOS 16.09

**Comment by noisecrew** [comment]
I have the same problem with the Windows Manager Only setup. NO desktop environment.
LXappearance did show the mouse themes not after I copy some Cursor themes manually into ~/.icons. but still refuse to change the theme.
also I am not using XCURSOR_PATH environment variable.
if there that is the solution how should i implement it?

**Comment by aloiscochard** [comment]
This is a serious issue for me, as it's very hard to see the mouse cursor on a retina display with the default size.

If there is any temporary workaround to change the size/theme, please let me know, that would be very useful.

**Comment by ambrop72** [comment]
@aloiscochard The workaround is to add these to your configuration.nix (this is what my patch makes the default):
```
environment.variables.XCURSOR_PATH = "$HOME/.icons";
environment.profileRelativeEnvVars.XCURSOR_PATH = [ "/share/icons" ];
```

**Comment by taohansen** [comment]
@ambrop72 "error: The option value `environment.variables.profileRelativeEnvVars' in `/etc/nixos/configuration.nix' is not a string or list of strings."

**Comment by ambrop72** [comment]
@taohansen the error suggests you didn't enter it right,
it's environment.profileRelativeEnvVars
not environment.variables.profileRelativeEnvVars.

**Comment by taohansen** [comment]
@ambrop72 I've made your necessary changes which allowed a rebuild however my mouse cursor still does not adopt a new cursor theme.

**Comment by aloiscochard** [comment]
Just to say that I got my custom cursor working  by following @ambrop72 advice, thanks a lot for the help!

**Comment by taohansen** [comment]
My cursor is only custom in over some applications. Any of the Xfce applications launched as user display the old default but qutebrowser and emacs shows my new cursor. Here's my [configuration.nix](https://gist.github.com/taohansen/d15e1fe4674a286cb9bcd8e3378a9f23).

**Comment by taohansen** [comment]
Adding environment.sessionVariables.GTK_PATH = "${config.system.path}/lib/gtk-3.0:${config.system.path}/lib/gtk-2.0"; in conjunction with @ambrop72's two lines has fixed this issue for me.

**Comment by yegortimoshenko** [comment]
This has been fixed in 8a039ff219a9d7630679803c1cbf82c82741b801.

I can't reproduce the issue (tried `nix-env -iA nixos.vanilla-dmz`):

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.5, NixOS, 18.03.git.ab1d2a5 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"overlay, nixos-18.03pre122666.afe9649210c"`
 - channels(chronos): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by AdsonCicilioti** [comment]
> This has been fixed in 8a039ff219a9d7630679803c1cbf82c82741b801.
> 
> I can't reproduce the issue (tried `nix-env -iA nixos.vanilla-dmz`):
> 
>     * system: `"x86_64-linux"`
> 
>     * host os: `Linux 4.14.5, NixOS, 18.03.git.ab1d2a5 (Impala)`
> 
>     * multi-user?: `yes`
> 
>     * sandbox: `yes`
> 
>     * version: `nix-env (Nix) 1.11.16`
> 
>     * channels(root): `"overlay, nixos-18.03pre122666.afe9649210c"`
> 
>     * channels(chronos): `""`
> 
>     * nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

Same problem in Gnome.

My Icon Themes placed in `~/.icons` are listed and defined, but not applied.
